packages/fuels/tests/harness.rs

```diff
@@ -2265,3 +2268,355 @@ async fn can_handle_sway_function_called_new() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn can_call_no_arg_predicate_returns_true() -> anyhow::Result<(), anyhow::Error> {
+    let amount_to_predicate: u64 = 16;
+    let asset_id = AssetId::default();
+    let predicate_code =
+        std::fs::read("tests/test_projects/predicate_true/out/debug/predicate_true.bin")?;
+    let receiver_address: Address = LocalWallet::new_random(None).address();
+    let wallets = launch_custom_provider_and_get_wallets(
+        WalletsConfig::default(),
+        Some(Config {
+            predicates: true,
+            utxo_validation: true,
+            ..Config::local_node()
+        }),
+    )
```

Comment:
basically I think your best bet is to have a wallet with the predicate address inside, this way you can avoid doing the `provider` lookup everytime.